test/test_folders.py

```diff
@@ -0,0 +1,134 @@
+import unittest
+import os.path
+
+from tableaudocumentapi import Datasource
+import xml.etree.ElementTree as ET
+
+
+TEST_ASSET_DIR = os.path.join(
+    os.path.dirname(__file__),
+    'assets'
+)
+TEST_TDS_FILE = os.path.join(
+    TEST_ASSET_DIR,
+    'folder_test.tds'
+)
+
+
+class TestFieldChange(unittest.TestCase):
+
+    def setUp(self):
+        self.tds = Datasource.from_file(TEST_TDS_FILE)
+
+    def current_hash(self):
+        """ Return a hash of the current state of the XML.
+
+        Allows us to easily identify whether the underlying XML-structure
+        of a TDS-file has actually changed. Avoids false positives if,
+        for example, a fields value has changed but the XML hasn't.
+        """
+        return hash(ET.tostring(self.tds._datasourceTree.getroot()))
```

Comment:
I'm not opposed to this method, but in other tests we generally either 1) Save the changes to the document to disk, then reopen and check for expected or 2) we lave the object in memory and just access the expected.

1 has the benefit of verifying serialization and deserialization, but it's "slower" -- we've got tempfile examples in the library if you want to go down that path.

Checking the hash is probably a fine route as well @RussTheAerialist might have an opinion. I think it makes the tests a tiny bit harder to follow but not a huge deal.

If you do stick with the hashing strategy, please use `hashlib` and md5 or something similar -- the internal hash function isn't intended to be super strong for this kind of use.